reporter/src/main/kotlin/reporters/fossid/FossIdReporter.kt

```diff
@@ -0,0 +1,90 @@
+
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.runBlocking
+
+import org.ossreviewtoolkit.clients.fossid.FossIdRestService
+import org.ossreviewtoolkit.clients.fossid.generateReport
+import org.ossreviewtoolkit.clients.fossid.model.report.ReportType
+import org.ossreviewtoolkit.clients.fossid.model.report.SelectionType
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.utils.core.log
+
+class FossIdReporter : Reporter {
+    companion object {
+        /** Name of the configuration property for the server URL. */
+        const val SERVER_URL_PROPERTY = "serverUrl"
+
+        /** Name of the configuration property for the API key. */
+        const val API_KEY_PROPERTY = "apiKey"
+
+        /** Name of the configuration property for the user name. */
+        const val USER_PROPERTY = "user"
+
+        /** Name of the configuration property for the report type. Default is [ReportType.HTML_DYNAMIC]. */
+        const val REPORT_TYPE_PROPERTY = "reportType"
+
+        /**
+         * Name of the configuration property for the selection type.
+         * Default is [SelectionType.INCLUDE_ALL_LICENSES].
+         */
+        const val SELECTION_TYPE_PROPERTY = "selectionType"
+
+        /**
+         * Name of key in [ScanResult.additionalData] containing the scancode.
+         * TODO: It should be unified with [FossId.SCAN_CODE_KEY], without creating a dependency between scanner and
+         * reporter.
+         */
+        const val SCAN_CODE_KEY = "scancode"
+    }
+
+    override val reporterName = "FossId"
+
+    override fun generateReport(input: ReporterInput, outputDir: File, options: Map<String, String>): List<File> {
+        val serverUrl = options[SERVER_URL_PROPERTY]
+            ?: throw IllegalArgumentException("No FossID server URL configuration found.")
+        val apiKey = options[API_KEY_PROPERTY]
+            ?: throw IllegalArgumentException("No FossID API Key configuration found.")
+        val user = options[USER_PROPERTY]
+            ?: throw IllegalArgumentException("No FossID User configuration found.")
+        val reportType = options[REPORT_TYPE_PROPERTY]?.let { ReportType.valueOf(it) } ?: ReportType.HTML_DYNAMIC
+        val selectionType =
+            options[SELECTION_TYPE_PROPERTY]?.let { SelectionType.valueOf(it) } ?: SelectionType.INCLUDE_ALL_LICENSES
+
+        val service = FossIdRestService.createService(serverUrl)
+
+        return runBlocking {
+            input.ortResult.scanner?.results?.scanResults?.values?.flatten()?.mapNotNull { result ->
+                result.additionalData[SCAN_CODE_KEY]?.let { scanCode ->
+                    async {
+                        this@FossIdReporter.log.info { "Generating report for scan $scanCode." }
+                        service.generateReport(user, apiKey, scanCode, reportType, selectionType, outputDir).getOrNull()
```

Comment:
This silently swallows failures in report generation. Shouldn't we at least log them?